compiler-plugin/src/main/java/io/ballerina/stdlib/ai/plugin/OpenAPIGenerator.java

```diff
@@ -134,22 +132,34 @@ public void perform(SyntaxNodeAnalysisContext context) {
     }
 
     private void generateOpenAPISpec(SemanticModel semanticModel, ServiceDeclarationNode serviceNode,
-                                     SyntaxTree syntaxTree, Map<Integer, String> services, Path inputPath,
+                                     SyntaxTree syntaxTree, Map<Integer, String> services,
                                      Project project, Path outPath, List<Diagnostic> diagnostics) {
         Optional<Symbol> serviceSymbol = semanticModel.symbol(serviceNode);
         if (serviceSymbol.isEmpty() || !(serviceSymbol.get() instanceof ServiceDeclarationSymbol)) {
             return;
         }
         extractServiceNodes(syntaxTree.rootNode(), services, semanticModel);
-        OASGenerationMetaInfo.OASGenerationMetaInfoBuilder builder =
-                new OASGenerationMetaInfo.OASGenerationMetaInfoBuilder();
-        ServiceNode service = new ServiceDeclaration(serviceNode, semanticModel);
-        builder.setServiceNode(service).setSemanticModel(semanticModel)
-                .setOpenApiFileName(services.get(serviceSymbol.get().hashCode()))
-                .setBallerinaFilePath(inputPath).setProject(project);
-        OASResult oasResult = ServiceToOpenAPIMapper.generateOAS(builder.build());
-        oasResult.setServiceName(constructFileName(syntaxTree, services, serviceSymbol.get()));
-        writeOpenAPIYaml(outPath, oasResult, diagnostics);
+        ListenerVisitor listenerVisitor = extractListenersFromProject(project);
+        Set<ListenerDeclarationNode> listeners = listenerVisitor.getListenerDeclarationNodes();
+
+        OpenAPI chatServiceSchema = ChatServiceOpenAPISchema.generate();
+        ServersMapper serversMapper = new ServersMapper(chatServiceSchema, listeners, serviceNode, semanticModel);
+        serversMapper.setServers();
+
+        String fileName = constructFileName(syntaxTree, services, serviceSymbol.get());
+        writeOpenAPIYaml(outPath, chatServiceSchema, fileName, diagnostics);
+    }
+
+    public static ListenerVisitor extractListenersFromProject(Project project) {
+        ListenerVisitor listenerVisitor = new ListenerVisitor();
+        project.currentPackage().moduleIds().forEach((moduleId) -> {
```

Comment:
Here we are passing the same `listenerVisitor` to all the modules and collect all the listeners from all the modules to a single set of listener nodes where we loose the module information. Also when we get back the listener node by checking the name, there is a possibility of getting a different listener declaration due to duplicate listener names in different modules.